packages/grid/x-data-grid-pro/src/hooks/features/rowGrouping/gridRowGroupingInterfaces.ts

```diff
@@ -2,6 +2,13 @@ export type GridRowGroupingModel = string[];
 
 export interface GridRowGroupingState {
   model: GridRowGroupingModel;
+
+  /**
+   * Tracks the model on the last pre-processing
+   * Allows to check if we need to re-build the grouping columns when the grid upserts a column.
+   * TODO: Move outside of the state
```

Comment:
I prefixed it to be able to move it in a minor.
@m4theushw maybe doing the same for the cache of the rows would be nice. Even if I agree that in practice probably nobody will try to access it.